tools/samtools.wdl

```diff
@@ -552,6 +552,13 @@ task merge {
         # -1 because samtools uses one more core than `--threads` specifies
         let "n_cores -= 1"
 
+        bams=""
+        for file in ~{sep(" ", bams)}
+        do
+          ln -sf $file
+          bams+=" $(basename $file)"
+        done
```

Comment:
There's an unhandled edge case. Hypothetical workflow:
1. Split a big BAM into small BAMs of `n` reads each
    - For reasons, they are all named `tmp.bam`
2. Do something
3. Merge them

The WDL engine will store each `tmp.bam` in it's own directory to prevent clobbering. Each invocation of `ln -sf` is going to overwrite the prior symlink. At the end of the loop there will only be one symlink pointing to the last BAM processed (and `bams` will equal ` tmp.bam tmp.bam tmp.bam` and probably cause a confusing error in samtools)

I think a solution will need some back and forth. Probably best to continue this over Slack?